### DISCUSSION TITLE
Feature Request: Support lz4 compression in parquet files

### DISCUSSION THREAD

**Discussion by danthegoodman1** [duckdb/duckdb#8950] [Ideas]
It trades lower compression for greater performance. More performant than snappy.

**Comment by mrocklin** [comment]
Hi 👋 

I thought I'd try to add some weight behind this issue.  I was looking at profiles for Dask + Arrow's parquet reader and found that a lot of it was Snappy decompression.  Looking online it seems like LZ4 is just strictly better these days, especially for decompression speeds at similar compression ratios to Snappy.  

I suspect that the Snappy default is due mostly to history within Spark, when Snappy had a good Java implementation and LZ4 didn't.   Spark chose to default to Snappy and then we all followed Spark in this choice.  Times have changed!

I was going to change Dask Dataframe's default compression from Snappy to LZ4 (Spark now supports both, as do many systems) but then learned that DuckDB didn't support LZ4, and so I paused.  

I don't know a ton here and am not well-researched in this area, but if my understanding is right I'd love to see community movement away from Snappy.

Thanks for reading!

**Reply by danthegoodman1** [reply]
Thanks :)

**Comment by hannes** [comment]
PR is up here: https://github.com/duckdb/duckdb/pull/11220

**Reply by hannes** [reply]
merged
